test/example/SnapshotGateTest.js

```diff
@@ -0,0 +1,717 @@
+
+    // Cast Diamond to IBosonAccountHandler. Use this interface to call all individual account handlers
+    accountHandler = await ethers.getContractAt("IBosonAccountHandler", protocolDiamond.address);
+
+    // Cast Diamond to IBosonOfferHandler
+    offerHandler = await ethers.getContractAt("IBosonOfferHandler", protocolDiamond.address);
+
+    // Cast Diamond to IGroupHandler
+    groupHandler = await ethers.getContractAt("IBosonGroupHandler", protocolDiamond.address);
+
+    // Deploy the SnapshotGate example
+    sellerId = "1";
+    [snapshotGate] = await deploySnapshotGateExample(["SnapshotGateToken", "SGT", protocolDiamond.address, sellerId]);
+
+    // Deploy the mock tokens
+    [foreign20] = await deployMockTokens(["Foreign20"]);
+  });
+
+  // All supported Exchange methods
+  context("📋 SnapshotGate Methods", async function () {
+    beforeEach(async function () {
+      // Initial ids for all the things
+      offerId = "1";
+      groupId = "1";
+      agentId = "0"; // agent id is optional while creating an offer
+
+      // Create a valid seller
+      seller = mockSeller(operator.address, admin.address, clerk.address, treasury.address);
+      expect(seller.isValid()).is.true;
+
+      // Create a second seller
+      seller2 = mockSeller(operator2.address, operator2.address, operator2.address, operator2.address);
+      expect(seller2.isValid()).is.true;
+
+      // AuthToken
+      emptyAuthToken = mockAuthToken();
+      expect(emptyAuthToken.isValid()).is.true;
+
+      // VoucherInitValues
+      voucherInitValues = mockVoucherInitValues();
+      expect(voucherInitValues.isValid()).is.true;
+
+      // Create the seller
+      await accountHandler.connect(admin).createSeller(seller, emptyAuthToken, voucherInitValues);
+
+      // Create the second seller
+      await accountHandler.connect(operator2).createSeller(seller2, emptyAuthToken, voucherInitValues);
+
+      // Create a valid dispute resolver
+      disputeResolver = mockDisputeResolver(
+        operatorDR.address,
+        adminDR.address,
+        clerkDR.address,
+        treasuryDR.address,
+        false
+      );
+      expect(disputeResolver.isValid()).is.true;
+
+      // Create DisputeResolverFee array so offer creation will succeed
+      disputeResolverFees = [
+        new DisputeResolverFee(ethers.constants.AddressZero, "Native", "0"),
+        new DisputeResolverFee(foreign20.address, "Foriegn20", "0"),
+      ];
+
+      // Make empty seller list, so every seller is allowed
+      const sellerAllowList = [];
+
+      // Register and activate the dispute resolver
+      await accountHandler
+        .connect(adminDR)
+        .createDisputeResolver(disputeResolver, disputeResolverFees, sellerAllowList);
+      await accountHandler.connect(deployer).activateDisputeResolver(disputeResolver.id);
+
+      // Manufacture snapshot for upload
+      snapshot = []; // { holder : string; tokenId: string; amount: string }[]
```

Comment:
```suggestion
      snapshot = []; // { owner : string; tokenId: string; amount: string }[]
```

Reply:
When referring to the snapshot, we are referring to token owners as holders. This helps disambiguates contract owner and who the owner of the local NFTs are in ownerOf

Reply:
fixed